Merge tag 'release-17.4.1' into TDE_REL_17_STABLE (#166)
Percona-Lab/postgres: contrib/pg_tde/documentation/docs/functions.md

  If the provider already stores a key by that name, the function returns an error.
* If set to `false`, an existing principal key may be reused.

### pg_tde_set_global_principal_key

> Creates or rotates the principal key for the current database using the specified global key provider and key name.

Reviewer: Something is wrong here. Can we set a global principal key with the global provider for the _current_ db?
Should it be "Creates or rotates a global principal key using the specified global key provider and the key name. This key is used for global settings like WAL encryption" ?

Author: Good comments but this is a merge commit so not the place to fix this. Sorry! Let's fix all of these as a separate PR.

Creates or rotates the default principal key for the server using the specified key provider.

> The default key is automatically used as a principal key by any database that doesn't have a specific principal key already created when the first encrypted database object is created.

Reviewer: ```suggestion
The default key is automatically used as a principal key by any database that doesn't have an individual key provider and key configuration.
```

Reviewer: This part: already created when the first encrypted database object is created. is super confusing. Can we get rid of it without losing the sense?

Author: Good comments but this is a merge commit so not the place to fix this. Sorry! Let's fix all of these as a separate PR.
